test/integration/cloud_event.ts

```diff
@@ -302,4 +304,20 @@ describe('CloudEvent Function', () => {
       })
       .expect(204);
   });
+
+  it('returns a 500 if the function throws an exception', async () => {
+    functions.cloudEvent('testTypedCloudEvent', () => {
+      throw 'I crashed';
+    });
+
+    // invoke the function with a CloudEvent with a string payload
+    const server = getTestServer('testTypedCloudEvent');
+    await supertest(server)
+      .post('/')
+      .send(TEST_CLOUD_EVENT)
+      .expect(res =>
+        assert.deepStrictEqual(res.headers['x-google-status'], 'error')
+      )
+      .expect(500);
```

Comment:
Would be good to check that the exception message doesn't appear in the response body.